scrapy_zyte_api/providers.py

```diff
@@ -75,9 +82,25 @@ async def __call__(
         zyte_api_meta = crawler.settings.getdict("ZYTE_API_PROVIDER_PARAMS")
         if html_requested:
             zyte_api_meta["browserHtml"] = True
-        for item_type, kw in item_keywords.items():
-            if item_type in to_provide:
-                zyte_api_meta[kw] = True
+
+        for cls in to_provide:
+            cls_stripped = strip_annotated(cls)
+            kw = item_keywords.get(cls_stripped)
+            if not kw:
+                continue
+            zyte_api_meta[kw] = True
+            if not is_typing_annotated(cls):
+                continue
+            metadata = cls.__metadata__
+            if cls_stripped == Product:
+                for option in ExtractFrom:
+                    if option in metadata:
+                        product_options = zyte_api_meta.setdefault("productOptions", {})
+                        if "extractFrom" in product_options:
```

Comment:
This doesn't check if the previous value is the same, maybe we should do that instead.

Reply:
Alternatively, we should treat values from different sources with different priorities and use the top one instead of raising this error.